MJxPrep.js

```diff
@@ -317,17 +366,66 @@ if (window.MJxPrep) {
     var openCallParens = funcStart + funcName.length
     var closeCallParens = findClosingBrace(expr, openCallParens)
     var argsString = expr.substring(openCallParens + 1, closeCallParens)
-    var args = splitList(argsString)
-    var newExpr = expr.substring(0, funcStart) +
-      action(funcName, args) +
-      expr.substring(closeCallParens + 1)
 
-    // Recursively replace the remaining instances of 'funcName(<args>)'
-    return replaceFunctionCalls(newExpr, funcName, action, funcStart + 1)
+    // replace any function calls that appear inside the arguments
+    // this will bail out without further recursion if argsString has no
+    // function calls
+    var processedArgsString = replaceFunctionCalls(argsString, funcName, action)
+    var args = shallowListSplit(processedArgsString)
+
+    // perform the action
+    var finished = expr.substring(0, funcStart) + action(funcName, args)
+    var unfinished = expr.substring(closeCallParens + 1)
+
+    // Recursively process the unfinished string
+    return replaceFunctionCalls(finished + unfinished, funcName, action, finished.length + 1)
```

Comment:
Earlier implementation of `replaceFunctionCalls` could be made to enter an infinite loop (see below). This fixes that.

```js
            //0123456890
const expr = 'cat(1) + 1'
const action = (funcName, args) => 'gonna crash cat(1)'
replaceFunctionCalls(expr, 'cat', action)
```
The original expression closes at position 5. When `replaceFunctionCalls` calls itself the first time, it starts searching at postion 5+1, but that 'cat' has been pushed past that position, so it is encountered again ...